rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py

```diff
@@ -168,23 +169,23 @@ def defragment(self, message):
             self.protocol.log("debug", log_msg)
 
             # Reconstruct the message
-            reconstructed_msg = ""
-            for i in range(0,message["total"]):
-                reconstructed_msg += self.received_fragments[msg_id]["fragment_list"][i]
-
-            log_msg = "reconstructed original message:\n"
-            log_msg += reconstructed_msg
+            reconstructed_msg = ''.join(self.received_fragments[msg_id]["fragment_list"][0:message["total"]])
+            log_msg = ["reconstructed original message:\n"]
+            log_msg.extend(reconstructed_msg)
+            log_msg = ''.join(log_msg)
             self.protocol.log("debug", log_msg)
 
             duration = datetime.now() - now
 
             # Pass the reconstructed message to rosbridge
+            reconstructed_msg = ''.join(reconstructed_msg)
```

Comment:
Again, `reconstructed_msg` is already a string at this point, no need to `join()` again.